cmake/Dependencies.cmake

```diff
@@ -544,3 +544,19 @@ if (USE_ZSTD)
   add_subdirectory(${PROJECT_SOURCE_DIR}/third_party/zstd/build/cmake)
   set_property(TARGET libzstd_static PROPERTY POSITION_INDEPENDENT_CODE ON)
 endif()
+
+if (USE_ONNX)
+  SET(ONNX_NAMESPACE "onnx_c2")
+  set(TEMP_BUILD_SHARED_LIBS ${BUILD_SHARED_LIBS})
+  # We will build onnx as static libs and embed it directly into the binary.
+  set(BUILD_SHARED_LIBS OFF)
+  add_subdirectory(${PROJECT_SOURCE_DIR}/third_party/onnx)
+  include_directories(${ONNX_INCLUDE_DIRS})
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DONNX_NAMESPACE=${ONNX_NAMESPACE}")
+  list(APPEND Caffe2_DEPENDENCY_LIBS onnx_proto)
+  caffe2_interface_library(onnx onnx_library)
```

Comment:
@Yangqing I've changed to use `caffe_inerface_library` after recent change. However, it doesn't seem to do what we ask it to do (whole-link). Here is the gcc command issued 
```
/usr/lib64/ccache/c++  -fPIC  -Wno-deprecated -DONNX_NAMESPACE=onnx_c2 -std=c++11 -O2 -fPIC -Wno-narrowing -Wno-invalid-partial-specialization -g  -shared  -o python/caffe2_pybind11
_state.so CMakeFiles/caffe2_pybind11_state.dir/python/pybind_state.cc.o CMakeFiles/caffe2_pybind11_state.dir/python/pybind_state_dlpack.cc.o CMakeFiles/caffe2_pybind11_state.dir/pyt
hon/pybind_state_mkl.cc.o ../lib/libcaffe2.a /usr/lib64/libprotobuf.so /usr/lib64/libgflags.so /usr/lib64/libglog.so ../lib/libnnpack.a ../lib/libcpuinfo.a ../lib/libpthreadpool.a -
llmdb -lleveldb -lsnappy ../lib/libgloo.a -lpthread -Wl,--whole-archive -Wl,--no-whole-archive ../lib/libonnx_proto.a -pthread -lprotobuf ../lib/libonnx.a -lgcc_s -lgcc -ldl
```
Notice how `../lib/libonnx.a ` dropped out of the `-Wl,--whole-archive -Wl,--no-whole-archive` structure. Any ideas? 

cc: @bddppq

Reply:
Confirmed that https://github.com/caffe2/caffe2/pull/2087 fixed the issue! Thanks.